frontend/server/src/DAO/ACLs.php

```diff
@@ -10,4 +10,43 @@
  * @access public
  */
 class ACLs extends \OmegaUp\DAO\Base\ACLs {
+        /**
+     * Get all ACLs owned by a user along with their type, alias, and an empty users array.
+     *
+     * @param int $userId The user ID whose ACLs are being fetched.
+     * @return list<array{acl_id: int, type: string, alias: string, users: list<array<string, never>>}>
+     */
+    public static function getUserOwnedAclTypesWithAliases(int $userId): array {
+        $sql = '
+            SELECT a.acl_id, c.alias, "contest" AS type
+            FROM ACLs a
+            INNER JOIN Contests c ON c.acl_id = a.acl_id
+            WHERE a.owner_id = ?
+            UNION
+            SELECT a.acl_id, c.alias, "course" AS type
+            FROM ACLs a
+            INNER JOIN Courses c ON c.acl_id = a.acl_id
+            WHERE a.owner_id = ?
+            UNION
+            SELECT a.acl_id, p.alias, "problem" AS type
+            FROM ACLs a
+            INNER JOIN Problems p ON p.acl_id = a.acl_id
+            WHERE a.owner_id = ?
+            UNION
+            SELECT a.acl_id, g.alias, "group" AS type
+            FROM ACLs a
+            INNER JOIN Groups_ g ON g.acl_id = a.acl_id
+            WHERE a.owner_id = ?
+        ';
+        $params = [$userId, $userId, $userId, $userId];
+
+        $rows = \OmegaUp\MySQLConnection::getInstance()->GetAll($sql, $params);
+
+        return array_map(fn($row) => [
+            'acl_id' => intval($row['acl_id']),
+            'type' => strval($row['type']),
+            'alias' => strval($row['alias']),
+            'users' => [],
+        ], $rows);
```

Comment:
There's no need to assert the type of each element here if MySQL already knows each of them:
```suggestion
        return array_map(fn($row) => [
            'acl_id' => $row['acl_id'],
            'type' => $row['type'],
            'alias' => $row['alias'],
            'users' => [],
        ], $rows);
```
So instead, you can simply specify the return type of the `GetAll()` function like this:
```php
/** @var list<array{acl_id: int, alias: string, type: string}> */
```

Even better, why not return only the result of the query in this function? That way, you can remove the `users` field, which isn't actually used here.

Reply:
i have removed the type assertion.
But we still need the emtpy users array for each  acl type since we need to append the user in each of the acl.
If we do not initialised the empty users array here then we need to run a for each loop again in order to have an empty users array to each acl of the user